src/Algebra/Lattice/Replicated.hs

```diff
@@ -0,0 +1,67 @@
+{-# LANGUAGE DeriveDataTypeable  #-}
+{-# LANGUAGE DeriveFoldable      #-}
+{-# LANGUAGE DeriveFunctor       #-}
+{-# LANGUAGE DeriveGeneric       #-}
+{-# LANGUAGE DeriveTraversable   #-}
+{-# LANGUAGE FlexibleContexts    #-}
+{-# LANGUAGE Safe                #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeOperators       #-}
+module Algebra.Lattice.Replicated (
+    Replicated(..)
+  ) where
+
+import Prelude ()
+import Prelude.Compat
+
+import Algebra.Lattice
+import Algebra.PartialOrd
+
+import Control.DeepSeq       (NFData (..))
+import Control.Monad         (ap, liftM2)
+import Data.Data             (Data, Typeable)
+import Data.Hashable         (Hashable (..))
+import Data.Universe.Class   (Finite (..), Universe (..))
+import Data.Universe.Helpers (Natural, Tagged, retag, (+++))
+import GHC.Generics          (Generic, Generic1)
+
+import qualified Test.QuickCheck as QC
+
+--
+-- Replicated
+--
+
+-- | Replicated lattice according to another lattice. 
+-- | All corresponding elements of first lattice replicas are ordered according to second lattice.
+data Replicated a b = Replicated a b
+    deriving ( Eq, Ord, Show, Read, Data, Typeable, Generic, Functor, Foldable, Traversable
+            , Generic1
+            )
+
+instance (NFData a, NFData b) => NFData (Replicated a b) where
+    rnf (Replicated a b) = rnf (rnf a, rnf b) -- TODO
```

Comment:
See e.g. https://github.com/phadej/lattices/blob/ae6c37705caf40e1536b57c6a144d252f5f519b0/src/Algebra/Lattice/Lexicographic.hs#L73

```haskell
rnf a `seq` rnf b
```

Reply:
Thanks for the hint, corrected.